pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctCountThetaSketchAggregationFunction.java

```diff
@@ -526,28 +543,51 @@ private Predicate getPredicate(String predicateString) {
    * passed to this method to be used when evaluating the expression.
    *
    * @param postAggregationExpression Post-aggregation expression to evaluate (modeled as a filter)
+   * @param expressions list of aggregation function parameters
    * @param sketchMap Precomputed sketches for predicates that are part of the expression.
    * @return Overall evaluated sketch for the expression.
    */
-  private Sketch evalPostAggregationExpression(FilterContext postAggregationExpression,
-      Map<Predicate, Sketch> sketchMap) {
-    switch (postAggregationExpression.getType()) {
-      case AND:
-        Intersection intersection = _setOperationBuilder.buildIntersection();
-        for (FilterContext child : postAggregationExpression.getChildren()) {
-          intersection.update(evalPostAggregationExpression(child, sketchMap));
-        }
-        return intersection.getResult();
-      case OR:
-        Union union = _setOperationBuilder.buildUnion();
-        for (FilterContext child : postAggregationExpression.getChildren()) {
-          union.update(evalPostAggregationExpression(child, sketchMap));
+  private Sketch evalPostAggregationExpression(
+      final ExpressionContext postAggregationExpression,
+      final List<Predicate> expressions,
+      final Map<Predicate, Sketch> sketchMap) {
+    if (postAggregationExpression.getType() == ExpressionContext.Type.LITERAL) {
+      throw new IllegalArgumentException("Literal not supported in post-aggregation function");
+    }
+
+    if (postAggregationExpression.getType() == ExpressionContext.Type.IDENTIFIER) {
+      final Predicate exp =
+          expressions.get(extractSubstitutionPosition(postAggregationExpression.getLiteral()) - 1);
```

Comment:
Perhaps pre-substitition in the constructor would be better? For example, if the same $k arg is repeated multiple times, we might avoid the use of matcher using a temporary alias map?